src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexManager.cs

```diff
@@ -54,10 +59,19 @@ LuceneIndexSettingsService luceneIndexSettingsService
                 shellOptions.Value.ShellsApplicationDataPath,
                 shellOptions.Value.ShellsContainerName,
                 shellSettings.Name, "Lucene");
-
             Directory.CreateDirectory(_rootPath);
             _luceneAnalyzerManager = luceneAnalyzerManager;
             _luceneIndexSettingsService = luceneIndexSettingsService;
+
+            // Typical geospatial context
+            // These can also be constructed from SpatialContextFactory
+            _ctx = SpatialContext.GEO;
+
+            var maxLevels = 11; // Results in sub-meter precision for geohash
+
+            // TODO demo lookup by detail distance
+            // This can also be constructed from SpatialPrefixTreeFactory
+            _grid = new GeohashPrefixTree(_ctx, maxLevels);
```

Comment:
I feel like this should also be abstracted away as a service, and provide the method to create a RecursivePrefixTreeStrategy, and also the indexable geoPoint.

Reply:
Ok, so that we can use both methods. Fair enough.